File: contests/typical90/002/solution2.rb
N = gets.to_i

arr = ["(",")"]

def is_correct?(sc)
  n = 0
  sc.each do |c|
    return false if n < 0
    n += 1 if c == "("
    n -= 1 if c == ")"
  end
  n == 0 ? true : false
end

res = []
arr.repeated_permutation(N) do |bits|
  res << bits.join if is_correct?(bits)
end

res.sort!
res.each do |r|
  puts r
end

File: contests/ABC/ABC439/C/solution2.py
N = int(input())
res = [0] * (N+1)
n = int(N ** 0.5)
for y in range(2,n+1):
    for x in range(1,y):
        val = x*x + y*y
        if val > N:
            break
        res[val] += 1
res2 = []
for i in range(1,N+1):
    if res[i] == 1:
        res2.append(str(i))
print(len(res2))
str = " ".join(res2)
if len(res2) != 0:
    print(str)

